index.js

```diff
@@ -29,7 +29,7 @@ module.exports = function kindOf(val) {
 
   // functions
   if (type === 'function' || val instanceof Function) {
-    if (val.constructor.name.slice(0, 9) === 'Generator') {
+    if (typeof val.constructor.name !== 'undefined' && val.constructor.name.slice(0, 9) === 'Generator') {
```

Comment:
initially, just meant that `if (typeof val.constructor.name === 'string' && ` would be better, but later update my thought :D